src/client.rs

```diff
@@ -172,7 +176,23 @@ impl Client {
             return Ok(Response::new(request.uri, Status::Excluded, request.source));
         }
         let status = match request.uri {
-            Uri::Website(ref url) => self.check_website(&url).await,
+            Uri::Website(ref url) => match self.check_website(url).await {
+                Status::Ok(code) if self.prefer_https && url.scheme() == "http" => {
+                    let mut https_url = url.clone();
+                    https_url
+                        .set_scheme("https")
+                        .unwrap_or_else(|_| unsafe { std::hint::unreachable_unchecked() });
+                    if self.check_website(&https_url).await.is_success() {
+                        Status::Error(
+                            format!("Insecure URL when HTTPS is available: {}", &url),
```

Comment:
Could we provide an improvement suggestion to the user like this?

```rust
format!("An alternative secure URL is available. Consider replacing {} with {}.", &url, &https_url),
```

Reply:
Modified as your suggestion.